### PR TITLE
[FUSE-1424] [WIP] Improve the documentation for Web-hooks

### PR DESCRIPTION
**Pull request by ggonpereira**
- [x] Add Record Webhook Implementation Example

### CODE REVIEW

File: accessing-data/record-webhooks.mdx [hunk 1]
```diff
@@ -129,8 +129,94 @@ Body:
...
+{
+  import_slug: 'f50df76a-e5b9-4784-80ba-6ef27948e2ba',
+  record_slug: 'cb28e820-8af5-4f3a-b01b-87f2eb3f5014',
+  validation_type: 'error',
+  column_name: 'id',
+  issue_description: 'An error occurred for the ID column.',
+}
+```
+
+- `import_slug` can be accessed through `req.body.import_slug`.
```
Comment (mikefrederick) [inline accessing-data/record-webhooks.mdx]: Describe what the parameters are.

File: accessing-data/record-webhooks.mdx [hunk 2]
```diff
@@ -129,8 +129,94 @@ Body:
...
+- `record_slug` can be accessed from the current position of the `req.body.records` array, doing `req.body.records[index].slug`.
+- `validation_type` can be "error" or "warning".
+- `column_name` should be the exact `internal_key` you defined in your Template columns implementation that you want to attach the validation.
+- `issue_description` is the message you want to show in the UI after the webhooks are processed.
+
+Finally, we send a response from our webhook to Fuse in the request done, sending the batch status as completed and sending the `validation_issues`.
+
+```javascript
+router.post('/record-webhook', async (req, res) => {
```
Comment (mikefrederick) [inline accessing-data/record-webhooks.mdx]: Use the Code Groups UI to show code samples if different programming languages.
https://mintlify.com/docs/content/components/code-groups

If there are other areas of the docs that need this update please do it.

File: accessing-data/record-webhooks.mdx [hunk 3]
```diff
@@ -129,8 +129,94 @@ Body:
...
+  // Compare the signatures
+  if (receivedSignature !== expectedSignature) {
+    return res.status(401).send('Mismatched signatures');
+  }
+
+  // Validate the data
+  const issues = [];
+  const importSlug = req.body.import_slug;
+
+  for (const record of req.body.records) {
```
Comment (mikefrederick) [inline accessing-data/record-webhooks.mdx]: This is too much to read IMO. Let's change to this:

```
for (const record of req.body.records) {
  // do something to save record.data here

  // passing validation errors back to Fuse
  if(recordDataHasIssues) {
    issues.push({
     import_slug: importSlug,
        record_slug: record.slug,
        validation_type: 'error',
        column_name: 'first_name',
        issue_description: 'An validation message to show to the user'
      });
  }}
}
```

File: accessing-data/record-webhooks.mdx [hunk 4]
```diff
@@ -129,8 +129,94 @@ Body:
...
     }]
   }
 }
-```
\ No newline at end of file
+```
+
+## Record Webhook Implementation Example
+
+In this example using Javascript and Express, we create an endpoint in your backend that initially receives the token `http_x_fuse_signature_256` containing the signature generated by Fuse and submitted to you. \
```
Comment (mikefrederick) [inline accessing-data/record-webhooks.mdx]: Remove this entire code description and the code. Just write:

Find sample code below for implementing a record webhook inline and responding with validation issues.

**Review by mikefrederick — CHANGES_REQUESTED** [review]

File: accessing-data/record-webhooks.mdx (hunk 1 above)
Comment (ggonpereira) [reply accessing-data/record-webhooks.mdx]: image
Something like this? @mikefrederick

File: accessing-data/record-webhooks.mdx (hunk 4 above)
Comment (ggonpereira) [reply accessing-data/record-webhooks.mdx]: image
Having something like this? @mikefrederick

File: accessing-data/record-webhooks.mdx (hunk 2 above)
Comment (ggonpereira) [reply accessing-data/record-webhooks.mdx]: Done.

File: accessing-data/record-webhooks.mdx (hunk 3 above)
Comment (ggonpereira) [reply accessing-data/record-webhooks.mdx]: Done.

File: accessing-data/status-webhooks.mdx [hunk 5]
```diff
@@ -29,3 +29,357 @@ A status of `records_submitted` means that all records from an Import are curren
...
+      }
+    );
+
+    const recordsToStore = importRecords.data;
+
+    // Do what you need with the records. In this example, let's store in our database
+    // Use the implementation for your database! This is just an example
+    await importRepository.createRecords(recordsToStore);
+
+    // Stored and processed! Let's update the Import status inside Fuse
```
Comment (ggonpereira) [inline accessing-data/status-webhooks.mdx:80]: This seems to be a bug.
I mentioned this with Mike, but apparently, we can't update the status to `records_processed` using the API. Not sure if this was corrected or not, but when trying to do the API call in this example, we get this error message:
`data: { message: 'Import already processed.' }`

**Comment by camilobarraza-tomtom** [comment]
docs are easier to follow now

**Pull request merged**